great_expectations/agent/actions/action.py

```diff
@@ -0,0 +1,27 @@
+from abc import ABC, abstractmethod
+from typing import List
+
+from pydantic import BaseModel
+
+from great_expectations.agent.models import Event
+from great_expectations.data_context import CloudDataContext
+
+
+class CreatedResource(BaseModel):
+    type: str
+    id: str
+
+
+class ActionResult(BaseModel):
+    id: str
+    type: str
+    created_resources: List[CreatedResource]
```

Comment:
Should the `created_resources` ever be modified or appended to after creation?

If not I think we should have mypy enforce this by behavior by making this a `Sequence[CreatedResource]`
```python
from typing import Sequence

...

class ActionResult(BaseModel):
    id: str
    type: str
    created_resources: Sequence[CreatedResource]

```

Reply:
correct, `created_resources` is expected to be static. updated in [72d707b](https://github.com/great-expectations/great_expectations/pull/7882/commits/72d707b8e3879af37f412ef03fe14b74bd851985)